src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -386,63 +380,90 @@ void submitEventCreateOperation(final DataOperation evtOp, final String nodeAddr
+						Character.MAX_RADIX
 				);
 				writeEvtFuture = evtWriter.writeEvent(routingKey, evtItem);
 			} else {
 				writeEvtFuture = evtWriter.writeEvent(evtItem);
 			}
 			evtOp.startRequest();
 			writeEvtFuture.handle((returned, thrown) -> completeEventCreateOperation(evtOp, evtItem, thrown));
-		} catch(final ScopeCreateException e) {
+		} catch (final ScopeCreateException e) {
 			LogUtil.exception(Level.WARN, e.getCause(), "{}: failed to create the scope \"{}\"", stepId, e.scopeName());
 			completeOperation((O) evtOp, RESP_FAIL_CLIENT);
-		} catch(final StreamCreateException e) {
+		} catch (final StreamCreateException e) {
 			LogUtil.exception(
-				Level.WARN, e.getCause(), "{}: failed to create the stream \"{}\"", stepId, e.streamName()
+					Level.WARN, e.getCause(), "{}: failed to create the stream \"{}\"", stepId, e.streamName()
 			);
 			completeOperation((O) evtOp, RESP_FAIL_CLIENT);
-		} catch(final NullPointerException e) {
-			if(!isStarted()) { // occurs on manual interruption which is normal so should be handled
+		} catch (final NullPointerException e) {
+			if (!isStarted()) { // occurs on manual interruption which is normal so should be handled
 				completeOperation((O) evtOp, INTERRUPTED);
 			} else {
 				completeFailedOperation((O) evtOp, e);
 			}
-		} catch(final InterruptRunException e) {
+		} catch (final InterruptRunException e) {
 			completeOperation((O) evtOp, INTERRUPTED);
 			throw e;
-		} catch(final Throwable cause) {
+		} catch (final Throwable cause) {
 			completeFailedOperation((O) evtOp, cause);
 		}
 	}
 
 	void submitEventReadOperation(final DataOperation evtOp, final String nodeAddr) {
-		// TODO: Evgeny, issue SDP-50
+		val controllerURI = URI.create(uriSchema);
+		val path = evtOp.dstPath();
+		val scopeName = path.substring(0, path.indexOf("/"));
+		val streamName = path.substring(path.indexOf("/") + 1);
+		val readerGroup = UUID.randomUUID().toString().replace("-", "");
+		val readerGroupConfig = ReaderGroupConfig.builder()
+				.stream(Stream.of(scopeName, streamName))
+				.build();
+		@Cleanup val readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
+		readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig);
+
+		@Cleanup val clientFactory = ClientFactory.withScope(scopeName, controllerURI);
+		@Cleanup val reader = clientFactory.createReader("reader",
+				readerGroup,
+				new JavaSerializer<String>(),
+				ReaderConfig.builder().build());
+		System.out.format("Reading all the events from %s/%s%n", scopeName, streamName);
+		EventRead<String> event = null;
+		try {
+			for (event = reader.readNextEvent(readTimeoutMillis); null != event.getEvent(); ) {
+				if (event.getEvent() != null) {
+					System.out.format("Read event '%s'%n", event.getEvent());
+				}
+			}
+		} catch (ReinitializationRequiredException e) {
+			e.printStackTrace();
+		}
+		System.out.format("No more events from %s/%s%n", scopeName, streamName);
```

Comment:
"Let's replace this w/ Loggers.MSG.trace(...) call" as Andrey says. I think it should be applied here as well